nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/serialization/VersionedFlowSynchronizer.java

```diff
@@ -257,6 +257,10 @@ private void mapCompatibleBundles(final DataFlow proposedFlow, final ExtensionMa
         final Set<String> missingComponentIds = proposedFlow.getMissingComponents();
         final VersionedDataflow dataflow = proposedFlow.getVersionedDataflow();
 
+        if (dataflow == null) {
```

Comment:
I propose the usage of the `isFlowEmpty`method here, which covers more possible variations for an empty `dataFlow`. I tested it via standard NiFi and it worked properly during initial startup.

Reply:
changed it to use the recommended method